### ISSUE TITLE
All Example Configurations are extended with Config((site, here, up) => .....

### ISSUE THREAD

**Issue by shahzaibk23** [chipsalliance/rocket-chip#2726]
All given example Configurations are extended with class Config((site,here,up) => 

What does these parameters (site,here,up) actually doing? Can you define it in simpler terms?

I've found this Object Config in rocket-chip/api-config-chipsalliance/design/craft/src/config/Config.scala
Is this the Config class which every configuration is extended by?

**Comment by lsteveol** [comment]
https://github.com/chipsalliance/api-config-chipsalliance/blob/master/README.md has some info

There is also this paper by Henry Cook (it is referenced in several links on the wikis/readmes, but some time those are broken).
[ProductiveDesignofExtensibleOnChipMemoryHierarchies.pdf](https://github.com/chipsalliance/rocket-chip/files/5538302/ProductiveDesignofExtensibleOnChipMemoryHierarchies.pdf)

Check out Section 2.5. 

Happy learnings
